### PR TITLE
add debug flag for emitter to debug the generator

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by lirenhe — APPROVED** [review]

**Comment by m-nash** [comment]
Is this to be able to attach the debugger once it gets to the dotnet code in the generator?  If so LGTM.

**Comment by ArcturusZhang** [comment]
> Is this to be able to attach the debugger once it gets to the dotnet code in the generator? If so LGTM.

Yes, we already have the flag in the generator to enable it to attach, but we are not passing the flag into the generator from the emitter.
This PR makes sure that we could pass in the flag through the emitter

File: src/CADL.Extension/Emitter.Csharp/src/emitter.ts [hunk 1]
```diff
@@ -171,11 +171,15 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
...
+
+                const shouldDebug = options.debug ?? false;
+                let debugFlag = "";
+                if (shouldDebug) debugFlag = "--debug";
                 const command = `dotnet --roll-forward Major ${resolvePath(
                     options.csharpGeneratorPath
                 )} --project-path ${outputFolder} ${newProjectOption} --clear-output-folder ${
                     options["clear-output-folder"]
-                }`;
+                } ${debugFlag}`;
```
Comment (chunyu3) [inline src/CADL.Extension/Emitter.Csharp/src/emitter.ts:182]: It will add extra space at the end of the command if shouldDebug==false.

Comment (ArcturusZhang) [reply src/CADL.Extension/Emitter.Csharp/src/emitter.ts]: I moved the space into the value of variable `debugFlag`

**Comment by lirenhe** [comment]
I believe this has no impact for the SDK repo, could you confirm?

**Comment by ArcturusZhang** [comment]
> I believe this has no impact for the SDK repo, could you confirm?

definitely no change in SDK

**Review by archerzz — APPROVED** [review]

**Pull request merged**